Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -1497,14 +1501,23 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         }
     }
 
-    // Attempt to shut down the tunnel
-    // On iOS 16 and below, as a workaround, we rekey to force a 403 error so that the tunnel fails to restart
     @MainActor
     private func attemptShutdown() async {
+        let cancelTunnel = {
+#if os(macOS)
+            try? self.tokenStore.deleteToken()
+#endif
```

Comment:
Do we need to check for system extension vs app extension here? I think it should be fine as-is, but checking anyway.

Reply:
Well, if access is revoked it should be fine regardless.  The only risk I see is if we get false positives as access-revoked, but then the problem would still not be this code, if that makes sense.